src/toast/ops/polyfilter.py

```diff
@@ -0,0 +1,607 @@
+                        continue
+                    for igroup in range(ngroup):
+                        good = group_det == igroup
+                        t = templates[isample][good].copy()
+                        ccinv = np.dot(t.T, t)
+                        try:
+                            cc = np.linalg.inv(ccinv)
+                            coeff[isample, igroup] = np.dot(cc, proj[isample, igroup])
+                        except np.linalg.LinAlgError:
+                            coeff[isample, igroup] = 0
+                if comm is not None:
+                    comm.allreduce(coeff)
+                t_solve += time() - t1
+
+                t1 = time()
+
+                for igroup in range(ngroup):
+                    local_dets = obs.local_detectors
+                    good = np.zeros(len(local_dets), dtype=np.bool)
+                    for idet, det in enumerate(local_dets):
+                        if group_index[det] == igroup:
+                            good[idet] = True
+                    if not np.any(good):
+                        continue
+                    for isample in range(nsample):
+                        if np.all(coeff[isample, igroup] == 0):
+                            views.detdata[self.det_flags][iview][:, isample] \
+                                |= (good * self.poly_flag_mask).astype(
+                                    views.detdata[self.det_flags][0].dtype
+                                )
+                templates = np.transpose(
+                    templates, [1, 2, 0]
+                ).copy()  # ndet x nmode x nsample
+                coeff = np.transpose(coeff, [1, 2, 0]).copy()  # ngroup x nmode x nsample
+
+                for idet, det in enumerate(obs.local_detectors):
+                    if det not in detector_index:
+                        continue
+                    igroup = group_index[det]
+                    ind = detector_index[det]
+                    signal = views.detdata[self.det_data][iview][idet]
+                    for mode in range(nmode):
+                        signal -= coeff[igroup, mode] * templates[ind, mode]
+
+                t_clean += time() - t1
+
+            """
+            print(
+                "Time per observation: {:.1f} s\n"
+                "   templates : {:6.1f} s\n"
+                "    get_norm : {:6.1f} s\n"
+                "  apply_norm : {:6.1f} s\n"
+                "       solve : {:6.1f} s\n"
+                "       clean : {:6.1f} s".format(
+                    time() - t0, t_template, t_get_norm, t_apply_norm, t_solve, t_clean
+                ),
+                flush=True,
+            )
+            """
+
+        return
+
+    def _finalize(self, data, **kwargs):
+        return
+
+    def _requires(self):
+        req = {
+            "meta": list(),
+            "shared": [self.shared_flags],
+            "detdata": [self.det_data, self.det_flags],
+            "intervals": [self.view],
```

Comment:
We should only append the flag and interval names if they are not `None`.

Reply:
Done.